classes/Employee.php

```diff
@@ -318,21 +347,25 @@ public static function employeeExists($email)
     /**
      * Check if employee password is the right one
      *
-     * @param string $passwd Password
+     * @param string $passwordHash Password
+     *
      * @return bool result
      */
-    public static function checkPassword($id_employee, $passwd)
+    public static function checkPassword($idEmployee, $passwordHash)
     {
-        if (!Validate::isUnsignedId($id_employee) || !Validate::isPasswd($passwd, 8)) {
+        if (!Validate::isUnsignedId($idEmployee)) {
```

Comment:
@ cookie one

Yeah, it can go back. `Validate::isPasswd()` is rather vague. Does is validate the password hash or plaintext password?
It sets a minimum of 8 characters, which is only possible when a plaintext password is uses. Hashes have a fixed length. Looks like there previously was a bug to me.